pyramid_tm/__init__.py

```diff
@@ -2,14 +2,22 @@
 import transaction
 
 from pyramid.settings import asbool
-from pyramid.util import DottedNameResolver
 from pyramid.tweens import EXCVIEW
+from pyramid.util import DottedNameResolver
+import warnings
+import zope.interface
+
+try:
+    from pyramid_retry import IRetryableError
+except ImportError:  # pragma: no cover
+    IRetryableError = zope.interface.Interface
 
 from .compat import reraise
-from .compat import binary_type, text_type
+from .compat import text_
```

Comment:
I wonder if instead of carrying this compat around, we should just import it from `pyramid.compat`... but I have a feeling that is a larger change in the context of `pyramid_*`.